tools/demo-gui/demo-gui/sof_controller_engine.py

```diff
@@ -22,58 +22,63 @@ def initialize_device():
 
     try:
         output = subprocess.check_output(["aplay", "-l"], text=True, stderr=subprocess.DEVNULL)
-
         match = re.search(r"card (\d+):.*\[.*sof.*\]", output, re.IGNORECASE)
         if match:
             card_number = match.group(1)
             device_string = f"hw:{card_number}"
             print(f"Detected SOF card: {device_string}")
         else:
-            print("No SOF card found.")
             raise RuntimeError("SOF card not found. Ensure the device is connected and recognized by the system.")
 
         controls_output = subprocess.check_output(["amixer", f"-D{device_string}", "controls"], text=True, stderr=subprocess.DEVNULL)
-
         volume_match = re.search(r"numid=(\d+),iface=MIXER,name='(.*Master Playback Volume.*)'", controls_output)
         if volume_match:
             volume_control = volume_match.group(2)
             print(f"Detected Volume Control: {volume_control}")
         else:
-            print("Master GUI Playback Volume control not found.")
             raise RuntimeError("Volume control not found.")
 
         eq_match = re.search(r"numid=(\d+),iface=MIXER,name='EQIIR1\.0 eqiir_coef_1'", controls_output)
         if eq_match:
             eq_numid = eq_match.group(1)
             print(f"Detected EQ numid: {eq_numid}")
         else:
-            print("EQ control not found.")
             raise RuntimeError("EQ control not found.")
 
     except subprocess.CalledProcessError as e:
         print(f"Failed to run device detection commands: {e}")
         raise
+    except RuntimeError as e:
+        print(e)
+        raise
 
 def scale_volume(user_volume):
-    normalized_volume = user_volume / 100.0
-    scaled_volume = 31 * (math.sqrt(normalized_volume))
-    return int(round(scaled_volume))
+    try:
+        normalized_volume = user_volume / 100.0
+        scaled_volume = 31 * math.sqrt(normalized_volume)
+        return int(round(scaled_volume))
+    except Exception as e:
```

Comment:
why would the above raise an exception? Only if `normalized_volume < 0`?

Reply:
Yes, but it's kind of impossible.

Reply:
right, so maybe it isn't needed? Also, I'm not an expert in python, but @marc-hb pointed out in a few cases, that over broad exception catching might actually not be a very good idea. It has a potential of hiding errors instead of causing the program to fail to prompt a proper investigation of the problem, in particular when *all* exceptions are caught, not some specific ones, which we know that we want to react to in a non-default way

Reply:
> It has a potential of hiding errors instead of causing the program to fail to prompt a proper investigation of the problem, in particular when all exceptions are caught, not some specific ones, which we know that we want to react to in a non-default way

W0178

https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-caught.html

`pylint` is not always right but it is right most of the time and it should always be used. It also offers plenty of ways to silence specific warnings, try `git grep pylint` for examples.

Reply:
> > It has a potential of hiding errors instead of causing the program to fail to prompt a proper investigation of the problem, in particular when all exceptions are caught, not some specific ones, which we know that we want to react to in a non-default way
> 
> W0178
> 
> https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-caught.html
> 
> `pylint` is not always right but it is right most of the time and it should always be used. It also offers plenty of ways to silence specific warnings, try `git grep pylint` for examples.

Thanks @marc-hb.
I'll have a look!